tools/rpkg/R/backend-dbplyr__duckdb_connection.R

```diff
@@ -0,0 +1,392 @@
+  sql_translator <- pkg_method("sql_translator", "dbplyr")
+  build_sql <- pkg_method("build_sql", "dbplyr")
+  sql_expr <- pkg_method("sql_expr", "dbplyr")
+  sql_prefix <- pkg_method("sql_prefix", "dbplyr")
+  sql_cast <- pkg_method("sql_cast", "dbplyr")
+  sql_paste <- pkg_method("sql_paste", "dbplyr")
+  sql_aggregate <- pkg_method("sql_aggregate", "dbplyr")
+  sql_aggregate_2 <- pkg_method("sql_aggregate_2", "dbplyr")
+  win_aggregate <- pkg_method("win_aggregate", "dbplyr")
+  win_aggregate_2 <- pkg_method("win_aggregate_2", "dbplyr")
+  win_over <- pkg_method("win_over", "dbplyr")
+  win_current_order <- pkg_method("win_current_order", "dbplyr")
+  win_current_group <- pkg_method("win_current_order", "dbplyr")
+
+
+  base_scalar <- pkg_method("base_scalar", "dbplyr")
+  base_agg <- pkg_method("base_agg", "dbplyr")
+  base_win <- pkg_method("base_win", "dbplyr")
+
+  sql_variant(
+    sql_translator(
+      .parent = base_scalar,
+      as.raw = sql_cast("VARBINARY"),
+      `%%` = function(a, b) sql_expr((!!a-!!b*FLOOR(!!a/!!b))),
+      `%/%` = function(a, b) sql_expr(FLOOR(!!a/!!b)),
+      `^` = sql_prefix("POW", 2),
+      bitwOr = function(a, b) sql_expr((CAST((!!a) %AS% INTEGER)) | (CAST((!!b) %AS% INTEGER))),
+      bitwAnd = function(a, b) sql_expr((CAST((!!a) %AS% INTEGER)) & (CAST((!!b) %AS% INTEGER))),
+      bitwXor = function(a, b) sql_expr(XOR((CAST((!!a) %AS% INTEGER)), (CAST((!!b) %AS% INTEGER)))),
+      bitwNot = function(a) sql_expr(~ (CAST((!!a) %AS% INTEGER))),
+      bitwShiftL = function(a, b) sql_expr((CAST((!!a) %AS% INTEGER)) %<<% (CAST((!!b) %AS% INTEGER))),
+      bitwShiftR = function(a, b) sql_expr((CAST((!!a) %AS% INTEGER)) %>>% (CAST((!!b) %AS% INTEGER))),
+      log = function(x, base = exp(1)) {
+        if (isTRUE(all.equal(base, exp(1)))) {
+          sql_expr(LN(!!x))
+        } else
+        if (base == 10) {
+          sql_expr(LOG10(!!x))
+        } else
+        if (base == 2) {
+          sql_expr(LOG2(!!x))
+        } else {
+          sql_expr(LOG(!!x) / LOG(!!base))
+        }
+      },
+      log10 = sql_prefix("LOG10", 1),
+      log2 = sql_prefix("LOG2", 1),
+
+      # See https://github.com/duckdb/duckdb/issues/530 about NaN, infinites and NULL in DuckDB
+      # The following is how R functions for detecting those should behave:
+      # Function 	    Inf 	–Inf 	NaN 	NA
+      # is.finite() 	FALSE FALSE FALSE FALSE
+      # is.infinite() TRUE 	TRUE 	FALSE FALSE
+      # is.nan() 	    FALSE FALSE TRUE 	FALSE
+      # is.na() 	    FALSE FALSE TRUE 	TRUE
+      is.na = function(a) build_sql("(", a, " IS NULL OR PRINTF('%f',", a, ") = 'nan')"),
+      is.nan = function(a) build_sql("(", a, " IS NOT NULL AND PRINTF('%f',", a, ") = 'nan')"),
+      is.infinite = function(a) build_sql("(", a, " IS NOT NULL AND REGEXP_MATCHES(PRINTF('%f',", a, "),'inf'))"),
+      is.finite = function(a) build_sql("(NOT (", a, " IS NULL OR REGEXP_MATCHES(PRINTF('%f',", a, "),'inf|nan')))"),
```

Comment:
Do we even need the checks for `IS NULL` ? 

Adding space and a comment.

```suggestion
      # https://github.com/duckdb/duckdb/issues/3019
      is.na = function(a) build_sql("(", a, " IS NULL OR PRINTF('%f', ", a, ") = 'nan')"),
      is.nan = function(a) build_sql("(", a, " IS NOT NULL AND PRINTF('%f', ", a, ") = 'nan')"),
      is.infinite = function(a) build_sql("(", a, " IS NOT NULL AND REGEXP_MATCHES(PRINTF('%f', ", a, "), 'inf'))"),
      is.finite = function(a) build_sql("(NOT (", a, " IS NULL OR REGEXP_MATCHES(PRINTF('%f', ", a, "), 'inf|nan')))"),
```

Reply:
I think  I had to add `IS NULL` as otherwise the result could be `NULL` in addition to `TRUE/FALSE`. But I will check once more.